src/CSV.jl

```diff
@@ -16,16 +22,22 @@ using PooledArrays
 # SentinelArrays.jl allow efficient conversion from Vector{Union{T, Missing}} to Vector{T}
 # it also provides the MissingVector and ChainedVector array types
 using SentinelArrays
+using WeakRefStrings
+export PosLenString
 
 struct Error <: Exception
     msg::String
 end
 
 Base.showerror(io::IO, e::Error) = println(io, e.msg)
 
+# constants
+const SINGLE_THREADED_POOL_DEFAULT = 0.75
```

Comment:
this is % of unique values? Why do you think for single thread it makes sense to make it as high as 75%?

Reply:
I based it off a conversation we had elsewhere, where because single-threaded parsing is typically w/ smaller files, we could try to pool more columns by default, since that tends to lead to optimizations in downstream workflows. I don't really have a strong opinion on what we make this or if we should just keep it consistent in both cases.

Reply:
OK - this perspective makes sense. Especially that - at least now - single and multi-threaded rules are not consistent in general (as one is exact and other is heuristic only)

Reply:
What does @nalimilan think: too weird to have different pooling thresholds in single threaded vs. multithreaded? Should we just make them the same?

Reply:
Ok, that's fine by me; I'll update to have a single default pool threshold value. Also happy to include a bias correction if someone wants to work out the math there; I've kind of just loosely been relying on the fact that we'll do a "reasonable number" of samples per chunk, which are naturally equally distributed across the file, so assume it's fairly representative of the full file. As Bogumil suggested, we can also bump up the default `lines_to_check` to increase the strength of that assumption.

Reply:
> bias correction

We cannot reliably do this because we are not doing a random sampling but sequential sampling.
E.g. you can have the following data `repeat(1:10^6, 100)`. Then for sure you will not be able to detect if you should pool correctly no matter what bias correction you do (you will think all is unique while it is not).

Exactly because of this effect I voted for increasing the threshold to 1000. The reason is that having this value at least reduces the risks associated with the fact that we are not sampling randomly but sequentially.

Reply:
Right, so we have to opposite biases:
- due to sequential sampling, values are more likely to be equal than in the full data -> we underestimate the proportion of unique values
- due to relying on a sample to estimate the proportion for the full data, values are more likely to be unique than in the full data -> we overestimate the proportion of unique value

AFAICT both biases can be somewhat limited by sampling a relatively large proportion of rows. Using at least 1000 rows sounds like a good idea, but I even suggest using a proportion of rows by default, so that more rows are sampled for large datasets, which prevents the bias from increasing. For example, for 100M rows, you may get 1000 unique values in a sample of 1000 rows, but there may actually be 10000 unique values and the data is sorted in some way (and in that case we should pool).

Reply:
> but I even suggest using a proportion of rows by default

I agree, but I understand that @quinnj is not able to know number of rows upfront. Normally I would suggest something like e.g. `maximum(1000, div(nrow(file), 1000))` but I understand this is not possible.